// argparse/lib/action-ck.js
/**
 * class Action
 *
 * Base class for all actions
 * Do not call in your code, use this class only for inherits your own action
 *
 * Information about how to convert command line strings to Javascript objects.
 * Action objects are used by an ArgumentParser to represent the information
 * needed to parse a single argument from one or more strings from the command
 * line. The keyword arguments to the Action constructor are also all attributes
 * of Action instances.
 *
 * #####Alowed keywords:
 *
 * - `store`
 * - `storeConstant`
 * - `storeTrue`
 * - `storeFalse`
 * - `append`
 * - `appendConstant`
 * - `count`
 * - `help`
 * - `version`
 *
 * Information about action options see [[Action.new]]
 *
 * See also [original guide](http://docs.python.org/dev/library/argparse.html#action)
 *
 **/"use strict";var $$=require("./const"),Action=module.exports=function(t){t=t||{};this.optionStrings=t.optionStrings||[];this.dest=t.dest;this.nargs=t.nargs!==undefined?t.nargs:null;this.constant=t.constant!==undefined?t.constant:null;this.defaultValue=t.defaultValue;this.type=t.type!==undefined?t.type:null;this.choices=t.choices!==undefined?t.choices:null;this.required=t.required!==undefined?t.required:!1;this.help=t.help!==undefined?t.help:null;this.metavar=t.metavar!==undefined?t.metavar:null;if(!(this.optionStrings instanceof Array))throw new Error("optionStrings should be an array");if(this.required!==undefined&&typeof this.required!="boolean")throw new Error("required should be a boolean")};Action.prototype.getName=function(){return this.optionStrings.length>0?this.optionStrings.join("/"):this.metavar!==null&&this.metavar!==$$.SUPPRESS?this.metavar:this.dest!==undefined&&this.dest!==$$.SUPPRESS?this.dest:null};Action.prototype.isOptional=function(){return!this.isPositional()};Action.prototype.isPositional=function(){return this.optionStrings.length===0};Action.prototype.call=function(){throw new Error(".call() not defined")};
